### PR TITLE
Cypress Tests: Added test for updating an existing patient.

### LINKED ISSUE #4234: Cypress: Update Existing Patient

### PR DESCRIPTION
**Pull request by SahilSingh177**
### WHAT
Added test that validates whether the patient has been successfully updated by changing/adding the following details:
- Added data to the present health condition
- Added ongoing medication data
-  Added data to the allergies
- In any medical history: selected diabetes & heart disease and input data to their respective text boxes
- Added insurance details for the patient
- Updated emergency number and current address

 This is followed by checking the dashboard to confirm if the patient has been updated successfully.

## Proposed Changes

- Fixes #4234 
- Added Test for updating a patient.
- Added `data-testid="patient-details"` to identify patient's address and medical details.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Comment by nihal467** [comment]
@SahilSingh177 

- merge the latest develop branch, the name is already getting edited, and don't edit the address beside that do the below-mentioned process

- add the following edit to the patient page:
- [ ] add data to the present health condition
- [ ] add ongoing medication data
- [ ] add data to the allergies
- [ ] in any medical history: select diabetes & heart disease and input data to their respective text boxes
- [ ] add insurance details for the patient
- [ ] changes his emergency number and current address

**Comment by SahilSingh177** [comment]
Sure. I'll implement these changes.

**Comment by SahilSingh177** [comment]
Hi, @nihal467. I'm having trouble finding the insurance details of a patient in the dashboard. Can you please guide me on where I can locate this information? I need to ensure that the patient's details have been successfully updated.

**Comment by nihal467** [comment]
@SahilSingh177 Currently, the insurance details are not shown in dashboard, so just add detail of the insurance . no need to verify it

File: cypress/e2e/patient_spec/patient_crud.cy.ts [hunk 1]
```diff
@@ -94,7 +94,25 @@ describe("Patient Creation with consultation", () => {
 
   it("Edit the patient details", () => {
     cy.awaitUrl(patient_url + "/update");
-    cy.get("[data-testid=name] input").clear().type("Test E2E User Editted");
+    cy.get("[data-testid=name] input").clear().type("Test E2E User Edited");
+    cy.get("#emergency_phone_number-div button")
+      .filter(":nth-child(2)")
+      .click();
```
Comment (SahilSingh177) [inline cypress/e2e/patient_spec/patient_crud.cy.ts]: Calling this to clear the emergency phone number.

**Comment by SahilSingh177** [comment]
@nihal467, I have updated the PR with the requested changes. Please review them once and let me know if any further changes are required.

**Comment by SahilSingh177** [comment]
@nihal467, all the tests are passing now, kindly review them once.

**Comment by nihal467** [comment]
@SahilSingh177 #5579 The phone number fields will be changed once the PR is merged. After the merge, I will review your PR again and proceed with merging it on or before tomorrow EOD.

**Comment by nihal467** [comment]
@SahilSingh177 the new field is merged, can you update the test since its failing

**Comment by SahilSingh177** [comment]
Sure. I'll update the tests

**Comment by SahilSingh177** [comment]
@nihal467, I have fixed the test and it's passing `cypress run` but "Add cypress passed label on success" is failing due to HttpError. I'm not sure what might be causing this.

**Comment by nihal467** [comment]
@SahilSingh177 we will look into it, your PR looks good to me

**Review by nihal467 — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**